FactoryDesign/FactoryDesign/IRepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FactoryDesign
+{
+    public interface IRepository
+    {
+        void Add(IProduct product, int fare);
```

Comment:
name of the function doesn't let me know anything, is it supposed to add fare to the product.